### DISCUSSION TITLE
demux and extract rpu at the same time with dovi_tool.exe?

### DISCUSSION THREAD

**Discussion by Distearth** [quietvoid/dovi_tool#181] [Q&A] [answered]
noob question.
I've been trying various versions of "START /B /WAIT cmd /c ffmpeg -probesize 42M -i injected_output.hevc -c:v copy -vbsf hevc_mp4toannexb -f hevc - | dovi_tool -m 2 convert --discard - extract-rpu - -o RPU.bin" in a batch file but nothing seems to work.

My current working flow is:
START /B /WAIT cmd /c ffmpeg -probesize 42M -i input.mkv -c:v copy -vbsf hevc_mp4toannexb -f hevc - | dovi_tool  demux -

START /B /WAIT cmd /c ffmpeg -probesize 42M -i input.mkv -c:v copy -vbsf hevc_mp4toannexb -f hevc - | dovi_tool extract-rpu - -o RPU.bin

Do I even need to extract the rpu if I'm just remuxing BL_RPU.hevc & --no-video input.mkv  ?

Does demux and extract-rpu always need to be preformed separately?

Thanks,

**Comment by quietvoid** [comment, marked as answer]
> Do I even need to extract the rpu if I'm just remuxing BL_RPU.hevc & --no-video input.mkv ?

I don't know, probably not?

> Does demux and extract-rpu always need to be preformed separately?

Yes.
